test/test_main_invocations.py

```diff
@@ -0,0 +1,136 @@
+# Copyright 2024 Canonical Ltd.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import os
+from pathlib import Path
+from typing import Callable, Type
+from unittest.mock import Mock
+
+import pytest
+
+import ops
+
+Reset = Callable[[], None]
+
+
+def type_test_dummy(_arg: Callable[[Type[ops.CharmBase], bool], None]):
+    """Usage:
+    from somewhere import main
+    type_test_dummy(main)
+    """
+
+
+def type_test_negative(_arg: Callable[[], None]):
+    """Usage:
+    from somewhere import main
+    type_test_negative(main)  # type: ignore
+
+    The `reportUnnecessaryTypeIgnoreComment` setting is expected to kick up a fuss,
+    should the passed argument match the expected argument type.
+    """
+
+
+@pytest.fixture
+def reset(monkeypatch: pytest.MonkeyPatch, tmp_path: Path):
```

Comment:
Could this have a different name? I feel like the fixture should have a noun name - I expect the fixture to "reset" when the test(s) are finished anyway.

Reply:
done, thank you